lib/secure-hash.js

```diff
@@ -53,7 +53,7 @@ module.exports = function (base64, crypto, promise) {
      * @param {secureHash~config} [config]
      * @return {Promise.<string>}
      */
-    function secureHashAsync(data, config) {
+    function secureHashAsync(data, config, regularEncode) {
```

Comment:
I don't like this.  Not one bit.  Think of a new developer seeing our system.  Do they want regular encode?  Do they want "non-regular" encode?  Should this encode or not?

Maybe split encoding from ID generation?